### PR TITLE
[New] implement image component

### PR DESCRIPTION
**Pull request by alphaoreo**
⭐ **New Features:**

- Added Image Component
- Added tests
- Updated Code Coverage

### CODE REVIEW

File: src/components/Image/Image.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const styles = rounded === true ? 'img-rounded' : '';
+  const styles2 = roundedCircle === true ? 'img-circle' : '';
+  const styles3 = thumbnail === true ? 'img-thumbnail' : '';
+  const finalstyle = `${styles} ${styles2} ${styles3}`;
+
+  return (
+    <Image
+      src={src}
+      alt={alt}
+      className={finalstyle.trimStart().trimEnd()}
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:35]: why do we need the trim?

File: src/components/Image/Image.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  src?: string;
+  /** Sets the alternate text for the image */
+  alt?: string;
+}
+
+const ImageWET = ({
+  rounded = false,
+  roundedCircle = false,
+  thumbnail = false,
+  src = '',
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:22]: if you look at the style guide there are a few more options that we should integrate:
https://wet-boew.github.io/wet-boew-styleguide/design/images-en.html

File: src/components/Image/Image.test.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import ImageWET from '@components/Image';
+
+describe('Image Tests', () => {
+  test('Test image component', () => {
```
Comment (dynamic11) [inline src/components/Image/Image.test.tsx:8]: group all the tests for shapes under one "describe". we should have a chat about this to organize the tests a bit better.

Also note: Each test should only be testing one thing. For example if you want to test the 'alt' prop then it should have its own test. The shape 'rounded' should have its own test. we can talk about this as well :)

**Review by dynamic11 — CHANGES_REQUESTED** [review]
Great work! Please see my comments and let me know if you would like to chat :)

File: src/components/Image/Image.tsx (hunk 2 above)
Comment (alphaoreo) [reply src/components/Image/Image.tsx:22]: I will add these today

File: src/components/Image/Image.tsx (hunk 1 above)
Comment (alphaoreo) [reply src/components/Image/Image.tsx:35]: No it is unnecessary I will remove it

File: src/components/Image/Image.test.tsx (hunk 3 above)
Comment (alphaoreo) [reply src/components/Image/Image.test.tsx:8]: I will fix the tests and their grouping

File: src/components/Image/stories/Image.shapes.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+Several shape variations for images.
+
+### Default
+
+<Canvas>
+  <Story
+    name="Default"
+    args={{
+      src: 'https://media.istockphoto.com/photos/very-closeup-view-of-amazing-domestic-pet-in-mirror-round-fashion-is-picture-id1281804798?b=1&k=20&m=1281804798&s=170667a&w=0&h=HIWbeaP_cQSngCz7l9t3xwyE2eyzVgIy3K6xIqPhJQA=',
+      rounded: false,
```
Comment (dynamic11) [inline src/components/Image/stories/Image.shapes.stories.mdx:24]: i dont think you need to define it as false. it auto defaults to false

File: src/components/Image/Image.test.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        thumbnail
+        rounded
+        responsive
+        fullWidth
+        media
+        thumbnailTile
+      />
+    );
+    expect(screen.getByRole('img')).toHaveClass('img-circle');
+    expect(screen.getByRole('img')).toHaveClass('img-thumbnail');
```
Comment (dynamic11) [inline src/components/Image/Image.test.tsx:22]: I would test each of these separately but group them under the same describe. As an example of nested describe statements you can check out the tests for the button component (line 16)

File: src/components/Image/Image.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  /** Use to make larger images scale properly, and never exceed the width of the parent container */
+  responsive?: boolean;
+  /** Use to stretch a smaller image to the width of the container. */
+  fullWidth?: boolean;
+  /** Use to add any kind of content like headings, paragraphs, or buttons into a thumbnail, to create a thumbnail tile effect */
+  thumbnailTile?: boolean;
+  /** Use to layer text and images for media content like blog comments, tweets, and so forth. */
+  media?: boolean;
+  /** Sets the source of the image */
+  src?: string;
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:21]: i dont think this should be optional

File: src/components/Image/Image.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  /** Sets the alternate text for the image */
+  alt?: string;
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:23]: dont make 'alt' optional either for accessibility reasons

**Review by dynamic11 — CHANGES_REQUESTED** [review]
please see my new comments. Also, if you are not using the images please delete them so they dont take space in the repo

**Review by dynamic11 — APPROVED** [review]

File: src/components/Image/Image.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  const styles3 = thumbnail === true ? 'img-thumbnail' : '';
+  const styles4 = responsive === true ? 'img-responsive' : '';
+  const styles5 = fullWidth === true ? 'full-width' : '';
+  const styles6 = thumbnailTile === true ? 'thumbnail' : '';
+  const styles7 = media === true ? 'media-object' : '';
+  const finalstyle = `${styles} ${styles2} ${styles3} ${styles4} ${styles5} ${styles6} ${styles7}`;
+
+  return <Image src={src} alt={alt} className={finalstyle} {...rest} />;
+};
+
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:49]: we need to add something like the following line before the export Image.displayName = 'Image';

If you see other components that dont have it please open an issue and list them there. We need to do it for consistency.

Comment (alphaoreo) [reply src/components/Image/Image.tsx:49]: What does that do/ change?

Comment (dynamic11) [reply src/components/Image/Image.tsx:49]: this is so the correct component name is displayed by error messages and storybook also uses it

**Comment by dynamic11** [comment]
please fix merge conflicts so we can merge

File: src/components/Image/Image.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import Image from 'react-bootstrap/Image';
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:2]: rename this to ImageRB and your component to Image

File: src/components/Image/Image.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  const styles4 = responsive === true ? 'img-responsive' : '';
+  const styles5 = fullWidth === true ? 'full-width' : '';
+  const styles6 = thumbnailTile === true ? 'thumbnail' : '';
+  const styles7 = media === true ? 'media-object' : '';
+  const finalstyle = `${styles} ${styles2} ${styles3} ${styles4} ${styles5} ${styles6} ${styles7}`;
+
+  return <Image src={src} alt={alt} className={finalstyle} {...rest} />;
+};
+
+export default ImageWET;
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:50]: rename to 'Image'

File: src/components/Image/index.ts [hunk 11]
```diff
@@ -0,0 +1,3 @@
+import ImageWET from './Image';
+
+export default ImageWET;
```
Comment (dynamic11) [inline src/components/Image/index.ts:3]: rename to 'Image'

**Review by dynamic11 — CHANGES_REQUESTED** [review]
import the the react-bootstrap image component as 'ImageRB' and rename yours to just 'Image'

File: src/components/Image/stories/Image.shapes.stories.mdx [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+<Canvas>
+  <Story
+    name="Rounded Circle"
+    args={{
+      src: 'https://media.istockphoto.com/photos/very-closeup-view-of-amazing-domestic-pet-in-mirror-round-fashion-is-picture-id1281804798?b=1&k=20&m=1281804798&s=170667a&w=0&h=HIWbeaP_cQSngCz7l9t3xwyE2eyzVgIy3K6xIqPhJQA=',
+      roundedCircle: true,
+    }}
+>
+    {Template.bind({})}
+
```
Comment (dynamic11) [inline src/components/Image/stories/Image.shapes.stories.mdx:55]: delete line

File: src/components/Image/stories/Image.shapes.stories.mdx [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+<Canvas>
+  <Story
+    name="Thumbnail"
+    args={{
+      src: 'https://media.istockphoto.com/photos/very-closeup-view-of-amazing-domestic-pet-in-mirror-round-fashion-is-picture-id1281804798?b=1&k=20&m=1281804798&s=170667a&w=0&h=HIWbeaP_cQSngCz7l9t3xwyE2eyzVgIy3K6xIqPhJQA=',
+      thumbnail: true,
+    }}
+
+>
+
```
Comment (dynamic11) [inline src/components/Image/stories/Image.shapes.stories.mdx:70]: delete line

**Review by dynamic11 — CHANGES_REQUESTED** [review]
please see comments

File: src/components/Image/Image.tsx [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+  alt = '',
+  ...rest
+}: ImageProps) => {
+  const styles = rounded === true ? 'img-rounded' : '';
+  const styles2 = roundedCircle === true ? 'img-circle' : '';
+  const styles3 = thumbnail === true ? 'img-thumbnail' : '';
+  const styles4 = responsive === true ? 'img-responsive' : '';
+  const styles5 = fullWidth === true ? 'full-width' : '';
+  const styles6 = thumbnailTile === true ? 'thumbnail' : '';
+  const styles7 = media === true ? 'media-object' : '';
```
Comment (dynamic11) [inline src/components/Image/Image.tsx:44]: I would rename these to something more descriptive like: roundedStyle

File: src/components/Image/Image.tsx [hunk 15]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import ImageRB from 'react-bootstrap/Image';
+import '../../style.css';
+
+export interface ImageProps extends React.HTMLAttributes<HTMLImageElement> {
+  /** Sets image shape as rounded. */
+  roundedCornerStyle?: boolean;
+  /** Sets image shape as circle. */
+  circleStyle?: boolean;
```
[the remark this answers is not in the document]
Comment (Rizvida1) [reply src/components/Image/Image.tsx:9]: Ideally, you'd want to use the same prop names as WET or Bootstrap

**Review by dynamic11 — APPROVED** [review]

**Pull request merged**